amqp/src/main/scala/akka/stream/alpakka/amqp/scaladsl/AmqpFlow.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.scaladsl
+
+import akka.NotUsed
+import akka.stream.alpakka.amqp.{AmqpWriteSettings, WriteMessage}
+import akka.stream.alpakka.amqp.impl.AmqpPublishFlowStage
+import akka.stream.scaladsl.{Flow, Keep}
+import akka.util.ByteString
+
+object AmqpFlow {
+
+  /**
+   * Scala API: Creates an [[AmqpFlow]] that accepts (ByteString, passthrough) elements.
+   *
+   * This stage materializes to a `Future[Done]`, which can be used to know when the Flow completes, either normally
+   * or because of an amqp failure
+   */
+  def simple[O](settings: AmqpWriteSettings): Flow[(ByteString, O), O, NotUsed] =
+    Flow[(ByteString, O)]
+      .map { case (s, passthrough) => (WriteMessage(s, false, false), passthrough) }
+      .viaMat(apply[O](settings))(Keep.right)
+
+  /**
+   * Scala API:
+   *
+   * Connects to an AMQP server upon materialization and sends incoming messages to the server.
+   * Each materialized flow will create one connection to the broker. This stage sends messages to
+   * the queue named in the replyTo options of the message instead of from settings declared at construction.
+   *
+   * This stage materializes to a Future[Done], which can be used to know when the Flow completes, either normally
+   * or because of an amqp failure
+   */
+  def apply[O](settings: AmqpWriteSettings): Flow[(WriteMessage, O), O, NotUsed] =
```

Comment:
Akka has added `withContext` support which composed with `Flow[(in, passThrough), (out, passThrough), ..]`, so it would be better to emit tuples. An alternative is to push the path-through into `WriteMessage` as many other connectors do.

Is there anything interesting that should be emitted as a `WriteResult`? One option could be to let the stage emit futures that signal confirmation so it becomes a user concern to fail the stream.

Reply:
Oh I didn't saw that. 
But to me it doesn't seem like a flow need a Tuple2 in and out to be compatible with that. The [`Flow.asFlowWithContext` method](https://doc.akka.io/docs/akka/current/stream/operators/Flow/asFlowWithContext.html) takes functions in order to transform from/into Tuple2, so with the current signature I think it should be compatible. Or did I miss something ?
([Some other examples here](https://github.com/akka/akka/pull/25951/files#diff-5bc226683709f15a4d491c27f88e98e7))

> Is there anything interesting that should be emitted as a WriteResult? 

Hum I don't think of anything that would make sense as `WriteResult` for this Flow :/. 

> One option could be to let the stage emit futures that signal confirmation so it becomes a user concern to fail the stream.

For the upcoming `async` and `asyncUnordered`, it could make sense to return a `Future[Unit]` (or `Try[Unit]`), but for this one, I'm afraid it could break the message order:
If we return a failed Future (or Try) downstream, I think it lets the publish stage take in the next element. So technically, couldn't the publish of the next element be sent before we fail the Flow ?

Reply:
I agree that it doesn't really make sense in this simple ("blocking") variant to pass futures downstream. Also, if we were to do that in `async` and `asyncUnordered` there would be no difference at all between those two - reordering is a result of out-of-order confirmations, and `async` is meant to internally wait for those confirmations and dispatch messages downstream in initial ("memorized") order. 
Future-emitting flow looks like yet another possible asynchronous variant where user could arbitrarily handle confirmations, but I'm not sure if there is any real use case for that.